sonar-ballerina-plugin/src/main/java/io/ballerina/sonar/BallerinaChecks.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.ballerina.sonar;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BallerinaChecks {
+    static final List<String> CORE_RULES = new ArrayList<>();
+
+    static {
+        CORE_RULES.add("B1");
```

Comment:
Is this why we need to republish whenever a new rule is added?

Reply:
Yes, SonarQube requires the rule metadata files and the respective identifiers to [initialize it's rule repository](https://github.com/ballerina-platform/sonar-ballerina/blob/3a77053135375adffcf93fd2d50e6c9972d54fd6/sonar-ballerina-plugin/src/main/java/io/ballerina/sonar/BallerinaRulesDefinition.java#L54) when an application instance is started.

Without having the rule identifiers and their respective metadata files added, SonarQube will reject any issues that are passed from the scan tool that that do not have those core rule identifiers.